jobbergate-agent/tests/jobbergate/test_compute.py

```diff
@@ -0,0 +1,307 @@
+
+                            if key not in aggregated_data:
+                                aggregated_data[key] = default_measurements.copy()
+
+                            for measurement in measurement_names:
+                                value = random.random() * 100
+                                measure = InfluxDBMeasure(
+                                    **{
+                                        "time": current_time,
+                                        "host": f"host_{host}",
+                                        "job": str(job),
+                                        "step": str(step),
+                                        "task": str(task),
+                                        "value": value,
+                                        "measurement": measurement,
+                                    }
+                                )
+                                measures.append(measure)
+
+                                # Aggregate value
+                                aggregated_data[key][measurement] = value
+                    current_time += 10
+
+        # Create aggregated list
+        aggregated_list = cast(
+            list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+            [
+                (
+                    time,
+                    host,
+                    step,
+                    task,
+                    *(aggregated_data[(time, host, step, task)][measurement] for measurement in measurement_names),
+                )
+                for (time, host, step, task) in aggregated_data
+            ],
+        )
+
+        return measures, aggregated_list
+
+    return _generate_and_aggregate
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "num_points_per_measurement, num_hosts, num_jobs, num_steps, num_tasks",
+    [
+        (1, 1, 1, 1, 1),
+        (3, 10, 1, 5, 10),
+        (7, 3, 1, 4, 2),
+    ],
+)
+async def test_aggregate_influx_measures__success(
+    num_points_per_measurement: int,
+    num_hosts: int,
+    num_jobs: int,
+    num_steps: int,
+    num_tasks: int,
+    generate_and_aggregate_job_metrics_data: Callable[
+        [int, int, int, int, int],
+        tuple[
+            list[InfluxDBMeasure],
+            list[tuple[int, str, str, str, float, float, float, float, float, float, float, float, float, float]],
+        ],
+    ],
+):
+    """
+    Test that the ``aggregate_influx_measures()`` function can successfully aggregate
+    a list of InfluxDBMeasure data points.
+    """
+    measures, expected_aggregated_data = generate_and_aggregate_job_metrics_data(
+        num_points_per_measurement, num_hosts, num_jobs, num_steps, num_tasks
+    )
+
+    start_time = time.monotonic()
+    aggregated_data = aggregate_influx_measures(iter(measures))
+    end_time = time.monotonic()
+
+    print(f"Aggregated {len(measures)} data points in {end_time - start_time:.5f} seconds")
```

Comment:
Leftover print?

Reply:
nope, it's for development purposes

Reply:
What is the purpose of leaving it in the code? If it needs to be here, shouldn't it be a logge rline instead? @fschuch what are your thoughts on this?

Reply:
The purpose is for the developer to benchmark the implementation. The repo doesn't have any official benchmarking framework so this is was the way I found of measuring my implementation

Reply:
I'm not opposed to adding benchmarking. I just don't think it belongs in the unit tests.

Reply:
at what moment in the development the developer would benchmark its code then?

Reply:
Perhaps an axillary script could be added to benchmark the method against a fixed set of test data. I don't think adding `print()` in unit tests for this is the right approach.

Reply:
let's close this conversion. `print` removed in [ab37a46](https://github.com/omnivector-solutions/jobbergate/pull/663/commits/ab37a463bc292a4f92680d39f30a5dd42d2eddfb).